lib/pbench/test/unit/server/query_apis/test_query_builder.py

```diff
@@ -0,0 +1,240 @@
+from pbench.server.api.resources import Schema
+import pytest
+from pbench.server.api.auth import Auth
+from pbench.server.api.resources.query_apis import ElasticBase, JSON, NoSelectedDatasets
+from pbench.server.database.models.users import User
+
+
+class TestQueryBuilder:
```

Comment:
I think the full matrix has at least 27 cases:
- The requestor is unauthorized
- The requestor is authorized but not admin
- The requestor is an admin

for which the request might target
- user is self
- user is some other user
- user is omitted

and for which the request might specify
- access public
- access private
- access omitted

(And, I don't know whether we need/want to test the difference between access/user omitted vs specified as `None`/`null`; also, I don't know whether we want to test the difference between targeting a non-existent but non-null user, as well, which potentially adds between 1 and 9 additional cases.)

I think you have 9 of the cases so far; I wonder if you can cover the rest of the matrix by parameterizing the existing scenarios.

Reply:
Yes, I debated full black-box testing, as you suggest here, vs my semi-white-box driven by knowledge of which cases actually have differing logic.

Reply:
🎉 ❗